File: agents/f#/Controllers/AnswerController.fs
namespace f_.Controllers

open Microsoft.AspNetCore.Mvc
open Microsoft.Extensions.Logging
open f_

[<ApiController>]
[<Route("[controller]")>]
type AnswerController (logger : ILogger<AnswerController>) =
    inherit ControllerBase()

    [<HttpGet>]
    member __.Get() : Answer =
        let rng = System.Random()

        { Name = "F#";
          Response = rng.Next(0,20);
          Children = [|
            { Name = "F♭";
              Response = rng.Next(0,20); 
              Children = [|
                { Name = "F";
                  Response = rng.Next(0,20);
                  Children = null; }
              |]; }
          |]}

File: agents/c#/Controllers/AnswerController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Logging;

namespace c_.Controllers
{
    [ApiController]
    [Route("[controller]")]
    public class AnswerController : ControllerBase
    {
        [HttpGet]
        public IActionResult Get()
        {
            var rng = new Random();

            return Ok(new { 
                Name = "C#",
                Response = rng.Next(0,20),
                Children = new [] {
                    new {
                        Name = "C♭",
                        Response = rng.Next(0,20),
                        Children = new [] {
                            new {
                                Name = "C",
                                Response = rng.Next(0,20)
                            }
                        }
                    }
                }
            });
        }
    }
}
